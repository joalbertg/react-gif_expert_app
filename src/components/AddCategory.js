import React, {useState} from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {
  const [value, setValue] = useState('');
  const handleAdd = event => {
    setValue(event.target.value);
  }

  const handleSubmit = event => {
    event.preventDefault();

    if(value.trim().length > 0) {
      setCategories(categories => [...categories, value]);
      setValue('');
    }
  }

  return(
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={value}
        onChange={handleAdd}
        placeholder='Add here...'
      />
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;

