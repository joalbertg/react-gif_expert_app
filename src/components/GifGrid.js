import React from 'react';

import { useFetchGifs } from '../hooks/useFetchGifs';

import GifGridItem from './GifGridItem';

const GifGrid = ({category}) => {
  const {data: gifs, loading} = useFetchGifs(category);

  return(
    <div className="card-grid">
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      { loading && <p className="loading animate__animated animate__flash">Loading</p> }
      {
        gifs.map(gif => (
          <GifGridItem
            key={gif.id}
            {...gif}
          />)
        )
      }
    </div>
  );
}

export default GifGrid;

