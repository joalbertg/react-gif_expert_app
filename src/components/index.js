export { default as AddCategory } from './AddCategory';
export { default as ListCategories } from './ListCategories';
export { default as GifGrid } from './GifGrid';

