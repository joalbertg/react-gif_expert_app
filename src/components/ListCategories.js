import React from 'react';
import PropTypes from 'prop-types';

import GifGrid from './GifGrid';

const ListCategories = ({categories}) => {
  return(
    <>
      { categories && categories.length !== 0 ?
          categories.map(category => (
            <GifGrid
              key={category}
              category={category}
            />
          ))
        :
        <h3>There are no categories to display.</h3>
      }
    </>
  );
}

ListCategories.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string).isRequired
}

ListCategories.defaultProps = {
  categories: []
}

export default ListCategories;

