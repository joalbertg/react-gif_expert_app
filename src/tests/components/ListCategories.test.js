import React from 'react';
import { shallow } from 'enzyme';

import { ListCategories } from '../../components';

describe('Tests ListCategories component', () => {
  test('Should display correctly', () => {
    const wrapper = shallow(<ListCategories />);
    const h3 = wrapper.find('h3').text();
    const gifGrids = wrapper.find('GifGrid');

    expect(wrapper).toMatchSnapshot();
    expect(h3).toBe('There are no categories to display.');
    expect(gifGrids).toEqual({});
    expect(gifGrids.exists()).toBe(false);
  });

  test('Should have an array of categories with 2 items', () => {
    const categories = ['Dragon Ball', 'Samurai X'];
    const wrapper = shallow(<ListCategories categories={categories}/>);
    const gifGrids = wrapper.find('GifGrid');
    const currentCategories = gifGrids.map(gifGrid => gifGrid.prop('category'));

    expect(wrapper).toMatchSnapshot();
    expect(gifGrids).toBeTruthy();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
    expect(currentCategories).toEqual(categories);
  })
});

