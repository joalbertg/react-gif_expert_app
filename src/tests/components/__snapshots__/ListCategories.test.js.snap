// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests ListCategories component Should display correctly 1`] = `
<Fragment>
  <h3>
    There are no categories to display.
  </h3>
</Fragment>
`;

exports[`Tests ListCategories component Should have an array of categories with 2 items 1`] = `
<Fragment>
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
  <GifGrid
    category="Samurai X"
    key="Samurai X"
  />
</Fragment>
`;
